zeebe/zb-db/src/main/java/io/camunda/zeebe/db/impl/rocksdb/transaction/RawTransactionalColumnFamily.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.db.impl.rocksdb.transaction;
+
+import static io.camunda.zeebe.util.buffer.BufferUtil.startsWith;
+
+import io.camunda.zeebe.db.TransactionContext;
+import io.camunda.zeebe.db.impl.DbBytes;
+import io.camunda.zeebe.protocol.ZbColumnFamilies;
+import org.agrona.concurrent.UnsafeBuffer;
+import org.rocksdb.ReadOptions;
+import org.rocksdb.RocksIterator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RawTransactionalColumnFamily {
+  private static final Logger LOG = LoggerFactory.getLogger(RawTransactionalColumnFamily.class);
+
+  protected final ZeebeTransactionDb<ZbColumnFamilies> transactionDb;
+  protected final ZbColumnFamilies columnFamily;
+  protected final ColumnFamilyContext columnFamilyContext;
+  protected final TransactionContext context;
+
+  public RawTransactionalColumnFamily(
+      final ZeebeTransactionDb<ZbColumnFamilies> transactionDb,
+      final ZbColumnFamilies columnFamily,
+      final TransactionContext context) {
+    this.transactionDb = transactionDb;
+    this.columnFamily = columnFamily;
+    columnFamilyContext = new ColumnFamilyContext(columnFamily.getValue());
+    this.context = context;
+  }
+
+  /**
+   * Run the given visitor for each key/value pair in the column family.
+   *
+   * @param visitor to run for each key/value pair: the key bytearray is "raw", i.e. contains also
+   *     the column family prefix. In order to get access to the key without the prefix, you need to
+   *     use {@link ColumnFamilyContext#wrapKeyView(byte[])}
+   */
+  public void forEach(final Visitor visitor) {
```

Comment:
:wrench: I know you likely copied this, but can't we break this down? That's a lot of indentation, and it feels like a good time to do it :shrug: